op-supervisor/supervisor/backend/db/logs/db.go

```diff
@@ -388,65 +556,32 @@ func (db *DB) searchCheckpoint(blockNum uint64, logIdx uint32) (entrydb.EntryIdx
 	return (i - 1) * searchCheckpointFrequency, nil
 }
 
-func (db *DB) AddLog(logHash types.TruncatedHash, block eth.BlockID, timestamp uint64, logIdx uint32, execMsg *types.ExecutingMessage) error {
-	db.rwLock.Lock()
-	defer db.rwLock.Unlock()
-	postState := logContext{
-		blockNum: block.Number,
-		logIdx:   logIdx,
-	}
-	if block.Number == 0 {
-		return fmt.Errorf("%w: should not have logs in block 0", ErrLogOutOfOrder)
-	}
-	if db.lastEntryContext.blockNum > block.Number {
-		return fmt.Errorf("%w: adding block %v, head block: %v", ErrLogOutOfOrder, block.Number, db.lastEntryContext.blockNum)
-	}
-	if db.lastEntryContext.blockNum == block.Number && db.lastEntryContext.logIdx+1 != logIdx {
-		return fmt.Errorf("%w: adding log %v in block %v, but currently at log %v", ErrLogOutOfOrder, logIdx, block.Number, db.lastEntryContext.logIdx)
-	}
-	if db.lastEntryContext.blockNum < block.Number && logIdx != 0 {
-		return fmt.Errorf("%w: adding log %v as first log in block %v", ErrLogOutOfOrder, logIdx, block.Number)
-	}
-	var entriesToAdd []entrydb.Entry
-	newContext := db.lastEntryContext
-	lastEntryIdx := db.lastEntryIdx()
-
-	addEntry := func(entry entrydb.Entry) {
-		entriesToAdd = append(entriesToAdd, entry)
-		lastEntryIdx++
-	}
-	maybeAddCheckpoint := func() {
-		if (lastEntryIdx+1)%searchCheckpointFrequency == 0 {
-			addEntry(newSearchCheckpoint(block.Number, logIdx, timestamp).encode())
-			addEntry(newCanonicalHash(types.TruncateHash(block.Hash)).encode())
-			newContext = postState
-		}
+func (db *DB) flush() error {
+	if err := db.store.Append(db.lastEntryContext.out...); err != nil {
+		return fmt.Errorf("failed to append entries: %w", err)
 	}
-	maybeAddCheckpoint()
+	db.lastEntryContext.out = db.lastEntryContext.out[:0]
+	db.updateEntryCountMetric()
+	return nil
+}
 
-	evt, err := newInitiatingEvent(newContext, postState.blockNum, postState.logIdx, logHash, execMsg != nil)
-	if err != nil {
-		return fmt.Errorf("failed to create initiating event: %w", err)
+func (db *DB) AddBlock(parentHash common.Hash, block eth.BlockID, timestamp uint64) error {
```

Comment:
We would want this function to be called in `ProcessLogs`, where an entire block of logs are processed, right?

That's the location where I'm now updating the local unsafe head, because the local unsafe head is effectively the last index in the database.

Reply:
Changed this, it's really "SealBlock", only called after all logs (if any) have been added.